src/lib/provable/gadgets/elliptic-curve.ts

```diff
@@ -444,14 +535,22 @@ function multiScalarMul(
       if (i % windowSize === 0) {
         // pick point to add based on the scalar chunk
         let sj = scalarChunks[j][i / windowSize];
-        let sjP =
-          windowSize === 1
-            ? points[j]
-            : arrayGetGeneric(
-                HashedPoint.provable,
-                hashedTables[j],
-                sj
-              ).unhash();
+        let sjP;
+        if (hashed) {
+          sjP =
+            windowSize === 1
+              ? points[j]
+              : arrayGetGeneric(
+                  HashedPoint.provable,
+                  hashedTables[j],
```

Comment:
it costs constraints to create the `hashedTables` (on line 521), which we only need in the deprecated case.

could you also add a check there and make it an empty array when `hashed = false`?

Reply:
[Fixed](https://github.com/o1-labs/o1js/pull/1669/commits/14c54ef5b807a8a18eff234d4a09028f23316588)